axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetDerogatoryLineMoveServiceImpl.java

```diff
@@ -0,0 +1,213 @@
+        || (incomeDepreciationAmount != null
+            && incomeDepreciationAmount.abs().compareTo(BigDecimal.ZERO) > 0)
+        || (derogatoryBalanceAmount != null
+            && derogatoryBalanceAmount.compareTo(BigDecimal.ZERO) > 0)) {
+
+      Account creditAccount = computeCreditAccount(fixedAssetDerogatoryLine);
+      Account debitAccount = computeDebitAccount(fixedAssetDerogatoryLine);
+      BigDecimal amount = computeAmount(fixedAssetDerogatoryLine);
+      generateMove(fixedAssetDerogatoryLine, creditAccount, debitAccount, amount);
+    }
+    fixedAssetDerogatoryLine.setStatusSelect(FixedAssetLineRepository.STATUS_REALIZED);
+    fixedAssetDerogatoryLineRepository.save(fixedAssetDerogatoryLine);
+
+    if (fixedAssetDerogatoryLine.getFixedAsset() != null) {
+      fixedAssetLineMoveService.realizeOthersLines(
+          fixedAssetDerogatoryLine.getFixedAsset(),
+          fixedAssetDerogatoryLine.getDepreciationDate(),
+          false,
+          true);
+    }
+  }
+
+  private BigDecimal computeAmount(FixedAssetDerogatoryLine fixedAssetDerogatoryLine) {
+    if (fixedAssetDerogatoryLine.getDerogatoryAmount().compareTo(BigDecimal.ZERO) > 0) {
+      return fixedAssetDerogatoryLine.getDerogatoryAmount().abs();
+    } else if (fixedAssetDerogatoryLine
+            .getIncomeDepreciationAmount()
+            .abs()
+            .compareTo(BigDecimal.ZERO)
+        > 0) {
+      return fixedAssetDerogatoryLine.getIncomeDepreciationAmount().abs();
+    }
+    return fixedAssetDerogatoryLine.getDerogatoryAmount().abs();
+  }
+
+  private Account computeDebitAccount(FixedAssetDerogatoryLine fixedAssetDerogatoryLine) {
+    FixedAsset fixedAsset = fixedAssetDerogatoryLine.getFixedAsset();
+    if (fixedAssetDerogatoryLine.getDerogatoryAmount().compareTo(BigDecimal.ZERO) > 0) {
+      return fixedAsset.getFixedAssetCategory().getExpenseDepreciationDerogatoryAccount();
+    } else if (fixedAssetDerogatoryLine
+            .getIncomeDepreciationAmount()
+            .abs()
+            .compareTo(BigDecimal.ZERO)
+        > 0) {
+      return fixedAsset.getFixedAssetCategory().getCapitalDepreciationDerogatoryAccount();
+    }
+    return fixedAsset.getFixedAssetCategory().getExpenseDepreciationDerogatoryAccount();
+  }
+
+  private Account computeCreditAccount(FixedAssetDerogatoryLine fixedAssetDerogatoryLine) {
+    FixedAsset fixedAsset = fixedAssetDerogatoryLine.getFixedAsset();
+    if (fixedAssetDerogatoryLine.getDerogatoryAmount().compareTo(BigDecimal.ZERO) > 0) {
+      return fixedAsset.getFixedAssetCategory().getCapitalDepreciationDerogatoryAccount();
+    } else if (fixedAssetDerogatoryLine
+            .getIncomeDepreciationAmount()
+            .abs()
+            .compareTo(BigDecimal.ZERO)
+        > 0) {
+      return fixedAsset.getFixedAssetCategory().getIncomeDepreciationDerogatoryAccount();
+    }
+    return fixedAsset.getFixedAssetCategory().getCapitalDepreciationDerogatoryAccount();
+  }
```

Comment:
Simplified the code, but did not factorized methods computeCreditAccount and computeDebitAccount, because both can don't return same accounts and accounts type can't be differenciated. (Could have added a parameter like a integer, string, or enum in factorized method but I don't think this is good idea) What do you think ?

Reply:
I see the issue, it might become more complex to factorize these methods than to leave them this way.